presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsTracker.java

```diff
@@ -139,6 +143,9 @@ else if (trackStatsForFailedQueries) {
             return ImmutableMap.of();
         }
 
+        HistoricalPlanStatisticsEntryInfo historicalPlanStatisticsEntryInfo = new HistoricalPlanStatisticsEntryInfo(
+                isNativeExecution ? HistoricalPlanStatisticsEntryInfo.WorkerType.CPP : HistoricalPlanStatisticsEntryInfo.WorkerType.JAVA, queryInfo.getQueryId());
+
```

Comment:
Can we get this `isNativeExecution` property via session directly? `queryInfo.getSession().toSession(sessionPropertyManager);` That way you wound not need to inject featuresConfig

Reply:
@jaystarshot Jay, we explicitly removed this property from the sessions because it doesn't make sense to allow this to be modified for individual queries. This is a cluster-wide property. I have a PR to actually delete it: #22183

CC: @tdcmeehan